documentation/modules/upgrading/proc-upgrade-kafka-kraft.adoc

```diff
@@ -0,0 +1,130 @@
+During this process, some brokers are using the old binaries while others have already upgraded to the new ones.
+Leaving the `metadataVersion` unchanged at the current setting ensures that the Kafka brokers and controllers can continue to communicate with each other throughout the upgrade.
+====
++
+For example, if upgrading from Kafka {KafkaVersionLower} to {KafkaVersionHigher}:
++
+[source,yaml,subs=attributes+]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    replicas: 3
+    metadataVersion: {KafkaMetadataVersionLower} # <1>
+    version: {KafkaVersionHigher} # <2>
+    # ...
+----
+<1> Metadata version is unchanged
+<2> Kafka version is changed to the new version.
++
+WARNING: You cannot downgrade Kafka if the `metadataVersion` for the new Kafka version is higher than the Kafka version you wish to downgrade to. 
+
+. If the image for the Kafka cluster is defined in `Kafka.spec.kafka.image` of the `Kafka` custom resource, update the `image` to point to a container image with the new Kafka version.
++
+See xref:con-versions-and-images-str[Kafka version and image mappings]
+
+. Save and exit the editor, then wait for the rolling updates to upgrade the Kafka nodes to complete.
++
+Check the progress of the rolling updates by watching the pod state transitions:
++
+[source,shell,subs=+quotes]
+----
+kubectl get pods my-cluster-kafka-0 -o jsonpath='{.spec.containers[0].image}'
+----
++
+The rolling updates ensure that each pod is using the broker binaries for the new version of Kafka.
+
+. Depending on your chosen xref:con-strategies-for-upgrading-clients-{context}[strategy for upgrading clients], upgrade all client applications to use the new version of the client binaries.
++
+If required, set the `version` property for Kafka Connect and MirrorMaker as the new version of Kafka:
++
+.. For Kafka Connect, update `KafkaConnect.spec.version`.
+.. For MirrorMaker, update `KafkaMirrorMaker.spec.version`.
+.. For MirrorMaker 2, update `KafkaMirrorMaker2.spec.version`.
++
+NOTE: If you are using custom images that are built manually, you must rebuild those images to ensure that they are up-to-date with the latest Strimzi base image. 
+For example, if you xref:creating-new-image-from-base-str[created a Docker image from the base Kafka Connect image], update the Dockerfile to point to the latest base image and build configuration.
+
+. Verify that the upgraded client applications work correctly with the new Kafka brokers.
+
+. If configured, update the Kafka resource to use the new `metadataVersion` version. Otherwise, go to step 9.
++
+For example, if upgrading to Kafka {KafkaVersionHigher}:
++
+[source,yaml,subs=attributes+]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    replicas: 3
+    metadataVersion: {KafkaMetadataVersionHigher}
+    version: {KafkaVersionHigher}
+    # ...
+----
```

Comment:
Now this is a bit tricky ... 

We should have a big warning here saying that Kafka support for downgrading the metadata version is limited and and might not be possible. And we should move here the warning part from line 80 that if you upgrade this and cannot downgrade, you might not be able to downgrade Kafka. Yet, at the same time we should make it clear that they have to eventually update the metadata version and cannot stick with the old one forever.

Reply:
added the following:

> WARNING: Exercise caution when changing the `metadataVersion`, as downgrading may not be possible. 
> You cannot downgrade Kafka if the `metadataVersion` for the new Kafka version is higher than the Kafka version you wish to downgrade to. 
> However, understand the potential implications on support and compatibility when maintaining an older version.